docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-in-states.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Deployment in states
+description: Deployment pipelines consist of three configurable stages.
+template: howto-guide-template
+originalLink: https://cloud.spryker.com/docs/deployment-pipelines
+originalArticleId: 14d91c9f-6c4e-4481-83ee-005683ce602f
+redirect_from:
+  - /docs/deployment-pipelines
+  - /docs/en/deployment-pipelines
+  - /docs/cloud/dev/spryker-cloud-commerce-os/deployment-pipelines/deployment-pipelines.html
+---
+
+When it comes to complex application, deploying to production environments is not just going from version one to version two. This document describes the states though which an application goes through during a deployment and how they affect its behavior.
+
+## Prerequisites
+
+To learn how pipelines work in SCCOS, see [Deployment pipelines](/docs/cloud/dev/spryker-cloud-commerce-os/configuring-deployment-pipelines/deployment-pipelines.html).
+
+## Production Pipeline
+
+Here is the list of the steps that we will find in a regular pipeline. Highlighted in **bold** are the steps that have a potential to break (or at least impact) our application.
+* Source
+* Please_approve
+* **Build_and_Prepare**
+* **Configure_RabbitMQ_Vhosts_and_Permissions**
+* **Run_pre-deploy_hook**
+* Pre_Deployment_Configuration
+* **Deploy_Scheduler**
+* **Run_install**
+* **Deploy_Spryker_services**
+* UpdateDeployedVersion
+* Post_Deployment_Configuration
+* **Run_post-deploy_hook**
+
+Let's see one by one how they can affect our application. Please keep in mind that this is a pessimistic approach, in order to show the potential issues within our deployments.
+It does not reflect the reality of a pipeline deployment.
+
+### Initial state
+This is how our application behaves  when no pipeline is running. In a fully working state:
+
+![Initial state](./images/initial_state/initial-state.gif)
+
+### Build_and_Prepare
```

Comment:
In the into, this step is identified as being able to affect the application. Can we describe how?

Reply:
Ah, this step doesn't really affect anything, but that's when we create the services we are going to replace later. It's kind of the context for what follows. Without explaining this step the rest would not make sense. 
But all by itself it doesn't break anything no.